### PR TITLE
Update ApplicationUnitTest to support multiple args for the same test.

### PR DESCRIPTION
**Pull request by KineticTheory**
### Background

+ Previously, we only allowed a single command line argument to be assigned during test registration.  This update allows options like '-a -b -c' to be specified for a single test.

### Description of changes

+ Apply new formatting rules for python (flake8) and cmake (cmake-format/cmake-lint).
+ Fix an indentation error in README.md.
+ Fix an issue related to 'draco_info' where the 'build date' was not captured correctly.

### Status

* Reference:
  * [Pre-Merge Code Review](https://github.com/lanl/Draco/wiki/Style-Guide)
  * [CDash Status](https://rtt.lanl.gov/cdash/index.php?project=Draco&filtercount=1&showfilters=1&field1=buildname&compare1=63&value1=-pr)
* Checks
  * [x] Travis/Appveyor CI checks pass
  * [x] Code coverage does not decrease
  * [x] [Valgrind test passes](https://rtt.lanl.gov/cdash3/index.php?project=Draco)
  * [x] [Toss3 checks pass](https://rtt.lanl.gov/cdash3/index.php?project=Draco)
  * [x] [Trinitite checks pass](https://rtt.lanl.gov/cdash3/index.php?project=Draco)
  * [ ] Code reviewed/approved, sufficient DbC checks, testing, documentation

### CODE REVIEW

File: config/application_unit_test.py [hunk 1]
```diff
@@ -1,770 +1,768 @@
...
+    re_mpiexec = re.compile(r"MPIEXEC_EXECUTABLE=([^\s]*)")
+    re_mpi_cores_per_cpu = re.compile(r"MPI_CORES_PER_CPU=([^\s]*)")
+    re_draco_info = re.compile(r"DRACO_INFO=([^\s]*)")
+    re_gdiff = re.compile(r"GDIFF=([^\s]*)")
+    re_pgdiff = re.compile(r"PGDIFF=([^\s]*)")
+
+    # run command can include spaces, modfiy regular expression for that
+    re_run_cmd = re.compile("RUN_CMD=(.*?)-D")
+
+    re_arg_value = re.compile("ARGVALUE=_av_(.*)_av_")
```
Comment (KineticTheory) [inline config/application_unit_test.py:150]: * This line is the only design change in this file.
* However, all of these regular expressions needed the `r` (raw string) qualifier to pass flake8 checks.

File: config/ApplicationUnitTest.cmake [hunk 2]
```diff
@@ -265,7 +265,7 @@ macro( aut_register_test )
   # Set arguments that don't change between python and CMake driver
   set (SHARED_ARGUMENTS
       -DAPP=${aut_APP}
-      -DARGVALUE=${argvalue}
+      -DARGVALUE=_av_${argvalue}_av_
```
Comment (KineticTheory) [inline config/ApplicationUnitTest.cmake:268]: @alexrlongne Can you think of a better way to do this.  I want to capture the string `-a -b -c` as a single argument.  I couldn't figure out how to make that work for both CMake and Python regex, so I added `_av_` as begin() and end() delimiters.

Comment (alexrlongne) [reply config/ApplicationUnitTest.cmake:268]: For python I can do something like this:
```
 arg = re.compile("(-[A-Za-z0-9]*)"
 my_str = "-option1 -option2"
 arg.findall(my_str)
```

and it returns a nice list:
```
['-option1', '-option2']
```

Let me try a few things in CMake and see if it has something nice like that.

Is there a simple interactive shell for cmake?  :)

Comment (alexrlongne) [reply config/ApplicationUnitTest.cmake:268]: Resolving this for now, we can revisit regexes later

**Review by alexrlongne — APPROVED** [review]

File: config/application_unit_test.py (hunk 1 above)
Comment (alexrlongne) [reply config/application_unit_test.py:150]: OK, that makes sense. Thanks for the explanation!

**Pull request merged**